### PR TITLE
SALTO-2833 Add change Validator to prevent the deploy of Case AssignmentRules with Case team

### PR DESCRIPTION
**Pull request by Uri-Bar**
_Add change Validator to prevent the deploy of Case AssignmentRules with Case team_

---
_Additional context for reviewer_:
None

---
_Release Notes_: 
Salesforce-adapter:

* Add change Validator to prevent the deploy of Case AssignmentRules with Case team
---

_User Notifications_: 
None

### CODE REVIEW

**Review by amitgoldarb — COMMENTED** [review]
This looks good.
Added a few very small comments.
Added a comment on the ticket though with a larger question.
Mainly, do teams exist on other types? Should we have that logic there as well?
What happens if the team exists in the target env? Are we skipping this change too aggresively?

File: packages/salesforce-adapter/src/change_validators/case_assignmentRules.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}).unknown(true)
+
+const isAssignmentRulesWithTeam = createSchemeGuard<AssignmentRules>(ASSIGNMENT_RULES)
+
+const createChangeError = (instance: InstanceElement): ChangeError => ({
+  elemID: instance.elemID,
+  severity: 'Error',
+  message: 'Deployment of case assignment rule that references a predefined case team is not supported. Please remove the ‘team’ property in order to deploy this element and then configure its team via the salesforce UI.',
+  detailedMessage: 'Deployment of case assignment rule that references a predefined case team is not supported in SF. Please  visit: https://help.salesforce.com/s/articleView?id=000387900&type=1',
+})
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/src/change_validators/case_assignmentRules.ts:62]: Just make sure @tomermevorach approves this before we merge.

**Review by amitgoldarb — APPROVED** [review]
LGTM
Only make sure that @tomermevorach approves this message.

File: packages/salesforce-adapter/src/change_validators/case_assignmentRules.ts (hunk 1 above)
Comment (tomermevorach) [reply packages/salesforce-adapter/src/change_validators/case_assignmentRules.ts:62]: looks good to me. Thx

**Pull request merged**